core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -218,6 +189,114 @@ public String toString() {
 
   protected abstract String defaultWarehouseLocation(TableIdentifier tableIdentifier);
 
+  protected class BaseMetastoreCatalogTableBuilder implements TableBuilder {
+    private final TableIdentifier identifier;
+    private final Schema schema;
+    private final ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
+    private PartitionSpec spec = PartitionSpec.unpartitioned();
+    private String location = null;
+
+    public BaseMetastoreCatalogTableBuilder(TableIdentifier identifier, Schema schema) {
+      Preconditions.checkArgument(isValidIdentifier(identifier), "Invalid table identifier: %s", identifier);
+
+      this.identifier = identifier;
+      this.schema = schema;
+    }
+
+    @Override
+    public TableBuilder withPartitionSpec(PartitionSpec newSpec) {
+      if (spec != null) {
+        this.spec = newSpec;
+      }
+      return this;
+    }
+
+    @Override
+    public TableBuilder withLocation(String newLocation) {
+      this.location = newLocation;
+      return this;
+    }
+
+    @Override
+    public TableBuilder withProperties(Map<String, String> properties) {
+      if (properties != null) {
```

Comment:
Same null question here

Reply:
We used to have this because overloaded createTable methods pass null for props:

```
Map<String, String> tableProperties = properties != null ? properties : Maps.newHashMap();
```

Since we are using the builder for the root `createTable` now, I added handling here too. For partition spec, it is optional but probably makes sense for consistency.